utils/validate_pr_quickstarts.js

```diff
@@ -0,0 +1,357 @@
+      path
+    )}/${screenshotFilename}`,
+  };
+};
+
+/**
+ * Creates the file path of each screenshot within the dashboard directory of a quickstart.
+ * @param {String} dashboardConfigPath - The file path of the config file within a quickstart's `dashboards` directory.
+ * @return {Array} A set of file-path strings for any screenshots within the same `dashboards` directory.
+ */
+const getQuickstartDashboardScreenshotPaths = (dashboardConfigPath) => {
+  const splitConfigPath = dashboardConfigPath.split('/');
+  splitConfigPath.pop();
+  const globPattern = `${splitConfigPath.join('/')}/*.+(jpeg|jpg|png)`;
+
+  return glob.sync(globPattern);
+};
+
+/**
+ * Builds input arguments for the `documentation` field.
+ * @param {Array} documentation - The documentation sections of a parsed quickstart config.yml file.
+ * @return {Array|undefined}  An set of objects that represent a quickstart's documentation in the context of a GraphQL mutation.
+ */
+const adaptQuickstartDocumentationInput = (documentation) =>
+  documentation &&
+  documentation.map((doc) => {
+    const { name, url, description } = doc;
+    return {
+      displayName: name,
+      url,
+      description,
+    };
+  });
+
+/**
+ * Reducer function that builds a set of unique quickstarts that were updated in a given PR.
+ * @param {Set} acc - A set of unique quickstarts being built by the reducer function.
+ * @param {Object} curr - A result from the GitHub API.
+ * @return {Set} A set of strings representing the unique quickstarts updated in a PR.
+ */
+const buildUniqueQuickstartSet = (acc, { filename }) => {
+  return getQuickstartFromFilename(filename)
+    ? acc.add(getQuickstartFromFilename(filename))
+    : acc;
+};
+
+/**
+ * Creates GraphQL requests to be used in the SubmitQuickstartMetadata mutation.
+ * @param {Array} files - A list of all files changed in the PR.
+ * @return {Array} Returns objects containing the file path of a quickstart's root config file and the variables used in the submitQuickstart mutation.
+ */
+const getGraphqlRequests = (files) => {
+  const uniqueQuickstarts = files.reduce(buildUniqueQuickstartSet, new Set());
+  const quickstartConfigPaths = getQuickstartConfigPaths(uniqueQuickstarts);
+
+  return quickstartConfigPaths.map((configPath) => ({
+    filePath: removeRepoPathPrefix(configPath),
+    variables: buildMutationVariables(readQuickstartFile(configPath)),
+  }));
+};
+
+const main = async () => {
+  const files = await fetchPaginatedGHResults(
+    GITHUB_API_URL,
+    process.env.GITHUB_TOKEN
+  ).catch((error) => {
+    throw new Error(`GitHub API returned: ${error.message}`);
+  });
+
+  const graphqlRequests = getGraphqlRequests(files);
+
+  const graphqlResponses = await Promise.all(
+    graphqlRequests.map(async ({ variables, filePath }) => {
+      const { data, errors } = await fetchNRGraphqlResults(
+        {
+          queryString: VALIDATE_QUICKSTART_MUTATION,
+          variables,
+        },
+        NR_API_URL,
```

Comment:
i moved these two (`NR_API_URL` and `NR_API_TOKEN`) into the fetchNRGraphqlResults function because i can't see where the input would be different than what is used in the `fetchNRGraphqlResults` but also i'm not sure what's best

Reply:
I saw that, I like it actually! I don't see any reason for those to be arguments. And anything to slim down this file is great 71cefbbc